### DISCUSSION TITLE
How to start the project with source code？

### DISCUSSION THREAD

**Discussion by piaoxiaobo** [coder/code-server#4374] [Q&A]
Hello everyone,
I am a front-end developer, and I want to deploy an online IDE in the near future. I want to know how to compile the source code of the code-server.

**Comment by aztack** [comment]
Follow the instructions on [CONTRIBUTING](https://coder.com/docs/code-server/latest/CONTRIBUTING)

**Comment by SeibertronSS** [comment]
> Follow the instructions on [CONTRIBUTING](https://coder.com/docs/code-server/latest/CONTRIBUTING)

@aztack 您好，我看了这个文档，可是当我用 yarn 命令编译的时候，会出现很多 ERROR，最后的服务也起不起来。

**Comment by aztack** [comment]
> > Follow the instructions on [CONTRIBUTING](https://coder.com/docs/code-server/latest/CONTRIBUTING)
> 
> @aztack 您好，我看了这个文档，可是当我用 yarn 命令编译的时候，会出现很多 ERROR，最后的服务也起不起来。

贴一下你的报错日志。
我在MacOS上需要先checkout v3.12.0。main分支上的代码运行报404错。

**Comment by piaoxiaobo** [comment]
@aztack 
![image](https://user-images.githubusercontent.com/34496544/137117750-16a33693-51ef-4b84-8679-5fb4e52bcd69.png)
![image](https://user-images.githubusercontent.com/34496544/137117481-df67a028-39a8-4f49-aa28-b8860d51e244.png)
![image](https://user-images.githubusercontent.com/34496544/137117591-5b7196a5-00e6-4e82-822c-eb90ab743a44.png)

执行命令：
git clone XXXX
git checkout v3.12.0
yarn
yarn watch
